server/src/main/java/org/elasticsearch/index/engine/LiveVersionMapArchiver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.util.BytesRef;
+
+/**
+ * Keeps track of the old map of a LiveVersionMap that gets evacuated on a refresh
+ */
+public interface LiveVersionMapArchiver {
+    /**
+     * Archive the old map evacuated due to a refresh
+     *
+     * @param old is the old map that is evacuated on a refresh
+     */
+    void afterRefresh(LiveVersionMap.VersionLookup old);
+
+    /**
+     * Trigger a cleanup of the archive based on the given generation
+     *
+     * @param generation the generation of the commit caused by the flush
+     */
+    void afterUnpromotablesRefreshed(long generation);
```

Comment:
I think we can avoid this method in the interface.

Reply:
@henningandersen I'd rather keep it here. The same way we have beforeRefresh/afterRefresh, I think having this in used in the live version map helps better understand  how the archiver works.

Reply:
i.e. it is another refresh-related hook that the live map version relies on.

Reply:
I've removed this from the interface.